test/Main.hs

```diff
@@ -62,6 +65,22 @@ bitVectorL = testL
 bitVectorR :: BitVector -> Small Int -> Property
 bitVectorR = testR
 
+data VectorRank = VectorRank [Word64] Position deriving (Eq, Show)
+
+data LenAsPos a = LenAsPos a deriving (Eq, Show)
```

Comment:
Since you only use this for the case where `a = VectorRank` you can just specialize it:

```haskell
data LenAsPos = LenAsPos VectorRank
```

Then you won't need the `FlexibleInstances` extension